lib/ResponsiveReactGridLayout.jsx

```diff
@@ -125,27 +125,27 @@ export default class ResponsiveReactGridLayout extends React.Component {
    * Width changes are necessary to figure out the widget widths.
    */
   onWidthChange(nextProps: typeof ResponsiveReactGridLayout.prototype.props) {
-    const {breakpoints, verticalLayout, verticalCompact, cols} = nextProps;
+    const {breakpoints, cols, verticalCompact} = nextProps;
     const newBreakpoint = nextProps.breakpoint || getBreakpointFromWidth(nextProps.breakpoints, nextProps.width);
 
     const lastBreakpoint = this.state.breakpoint;
 
     // Breakpoint change
     if (lastBreakpoint !== newBreakpoint || this.props.breakpoints !== breakpoints || this.props.cols !== cols) {
-
-      // Store the current layout
       const layouts = nextProps.layouts;
-      layouts[lastBreakpoint] = cloneLayout(this.state.layout);
 
-      // Find or generate a new one.
+      // Preserve the current layout if the current breakpoint is not present in the next layouts.
+      if (!(lastBreakpoint in layouts)) layouts[lastBreakpoint] = cloneLayout(this.state.layout);
```

Comment:
This is fine for now - likely to remove entirely though. If you want to keep layout changes that the user makes, it should be mandatory to listen to `onLayoutChange` and update your model.

I'll likely go with a `WidthProvider`-style HOC for preserving the old hands-off behavior with the default core behavior being stateless.